### ISSUE TITLE
Accessibility review

### ISSUE THREAD

**Issue by boonebgorges** [livinglab/webwork-for-wordpress#108]
We need an accessibility review of OL-WW. At a glance, it doesn't look awful - a few missing 'alt' attributes and form labels, as well as a couple of contrast issues. Some attention should also be paid to interactive areas and aria-live.

**Comment by boonebgorges** [comment]
I've finished an initial review.

The markup changes have been minimal. Avatars needed alt text, and a number of form elements (specifically, sidebar filters) needed labels.

In the interest of moving us along, I made a few judgment calls to get us compliant with contrast ratios. Screenshots follow. I'm happy to revisit any of these based on the feedback of @jennaspevack and others - I just thought it'd be easier to start with something concrete :-D 

"Top" link - Changed background from #668fcf (our "light blue") to #0044af (our "dark blue"). fc37f41
![screenshot_2017-12-14_14-42-08](https://user-images.githubusercontent.com/246627/34013580-a297fc44-e0dd-11e7-9d4e-34da5fa51b15.png)
![screenshot_2017-12-14_14-45-09](https://user-images.githubusercontent.com/246627/34013586-acb05604-e0dd-11e7-81ef-318a6e40b96b.png)

Footer links - Changed text to white, added underline to define as a link. 9dfd0f8
![screenshot_2017-12-14_14-37-19](https://user-images.githubusercontent.com/246627/34013620-d074fe00-e0dd-11e7-94a9-3bb6e1261f76.png)
![screenshot_2017-12-14_14-38-25](https://user-images.githubusercontent.com/246627/34013626-d65cecd8-e0dd-11e7-8ef9-194a6e8a6df5.png)

Course metadata: Nudged a shade darker to get WAVE to pass. c81f1ce
![screenshot_2017-12-14_14-35-38](https://user-images.githubusercontent.com/246627/34013657-fc6dfcbe-e0dd-11e7-9eb7-973da920451a.png)
![screenshot_2017-12-14_14-36-11](https://user-images.githubusercontent.com/246627/34013652-f7160afe-e0dd-11e7-82bf-0b49293d2ef0.png)

Disabled "vote" icon for logged-out user: Nudged a shade darker. 135321b
![screenshot_2017-12-14_14-32-18](https://user-images.githubusercontent.com/246627/34013699-257c1992-e0de-11e7-9602-b59cc464c05b.png)
![screenshot_2017-12-14_14-32-01](https://user-images.githubusercontent.com/246627/34013708-29b56e78-e0de-11e7-80d8-40220d816991.png)

**Comment by boonebgorges** [comment]
Everything on http://openlabdev.org/webwork-playground/ is now passing WAVE's AA check, with the exception of a single image imported from WeBWorK, which is missing an alt tag.

**Comment by boonebgorges** [comment]
(In logged-out state, that is.)

**Comment by jennaspevack** [comment]
Color/contrast changes look good to me. Thanks Boone!

**Comment by boonebgorges** [comment]
Thanks, Jenna!!

I'll leave this ticket open for any necessary WAVE testing from the team.

**Comment by bree-z** [comment]
I did a WAVE test, and see the following errors: 

* 9 images with missing alt text (all from questions/images posted this week).
* 1 missing form label (from [this question](http://openlabdev.org/webwork-playground/#:problemId=local/GraphingSineCosine/function-graphing-period.pg:questionId=11666), which has a dropdown at the bottom for the question "﻿Which graph matches the given function?")

I also see 3 contrast errors, all in the footer, which looks different for me than the screenshot above:

![screen shot 2018-01-05 at 10 31 52 am](https://user-images.githubusercontent.com/7189824/34615690-c6a9985e-f203-11e7-842f-b684c03e0e4d.png)

**Comment by boonebgorges** [comment]
Thanks, Bree - if I understand correctly, all of the current errors (missing alt text and missing form label) are errors in the source content on WeBWorK. As a general rule, there's not much we can do in an automated fashion to fix the markup that comes from WeBWorK - it must be fixed there instead.

In the case of problem images, I *could* generate some dummy text - something like "Image for problem xyz". This would fool the WAVE tool, though I'm unsure whether it'd solve any actual accessibility problems.

The contrast errors appear to be caused by the recent update to the Hemingway theme. I've reinstated the fixes in https://github.com/livinglab/openlab/commit/8e307a56d08b9b892e63f76a7e38e5a37226fe43, and the

**Comment by bree-z** [comment]
Thanks, Boone!  That's what I thought re content from WeBWorK, but thought I'd mention it just in case.  :)  

I don't know what's involved in fixing it on the WeBWorK side, but that seems like the more effective thing to work towards rather than generating dummy text; not sure what others think though.  

And, the contrast errors are all gone!
